### PR TITLE
Add automated versioning with inner gradle plugin

### PR DESCRIPTION
**Pull request by Joaquimmnetto**
## Description
Added a mini-plugin for gradle inside the buildSrc directory that deals with versioning for this repository. The plugins creates the `branchVersion` task, with the following possible configurations:

 * `updateType`: Kind of update that is being done. Should be "MAJOR", "MINOR", or "PATCH". Mandatory field. Can also be set through the property `version.updateType`.
*  `remote`: git remote name to where push changes. Defaults to "origin". Can also be set through the property `version.remote`.
*  `dryRun`: if the current run should be a "dry run", that is, changes are not actually pushed to remote. Boolean, defaults to `false`. Can also be set trough the property `version.dryRun`.

## Changes
* ![ADD] Automated version management for this project

[ADD]:https://resources.atlas.wooga.com/icons/icon_add.svg "Add"

### CODE REVIEW

**Review by Larusso — APPROVED** [review]
Looks good to me. I would only suggest to use the provider API so you get to know it better from a using point. I know that we don't need lazy evaluation here since our usecase is internal to this project only. It could help you to get more familiar with it.

File: Jenkinsfile [hunk 1]
```diff
@@ -15,16 +15,28 @@
...
+    parameters {
+        choice(name: 'releaseType', choices: ['NONE', 'MAJOR', 'MINOR', 'PATCH'],
+                description:  "Pick a release type, or NONE if this run shouldn't generate a release")
+    }
+
     agent any
     stages {
         stage("Check") {
-            steps { gradleWrapper "check" }
+            steps { gradleWrapper "check --info --stacktrace" }
```
Comment (Larusso) [inline Jenkinsfile:39]: Check out how we do this for the other pipelines. Essentially you need to add this to the pipeline parameters:

```groovy
choice(choices: ["", "quiet", "info", "warn", "debug"], description: 'Choose the log level', name: 'LOG_LEVEL')
booleanParam(defaultValue: false, description: 'Whether to log truncated stacktraces', name: 'STACK_TRACE')
booleanParam(defaultValue: false, description: 'Whether to refresh dependencies', name: 'REFRESH_DEPENDENCIES')
```

And then the gradleWrapper step takes the values from the env.

File: buildSrc/build.gradle [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    implementation gradleApi()
+    testImplementation platform("org.spockframework:spock-bom:1.2-groovy-2.4")
+    testImplementation "org.spockframework:spock-core"
+    testImplementation 'com.wooga.spock.extensions:spock-github-extension:0.1.2'
+    testImplementation ('com.netflix.nebula:nebula-test:[8,9)') {
+        version {
+            strictly '8.1.0'
+        }
+    }
+}
```
Comment (Larusso) [inline buildSrc/build.gradle:28]: missing newline

**Review by Larusso — CHANGES_REQUESTED** [review]

File: buildSrc/build.gradle (hunk 2 above)
Comment (Joaquimmnetto) [reply buildSrc/build.gradle:28]: Done.

File: Jenkinsfile (hunk 1 above)
Comment (Joaquimmnetto) [reply Jenkinsfile]: Done.

**Pull request merged**